packages/editor/src/components/post-title/index.js

```diff
@@ -60,6 +62,57 @@ class PostTitle extends Component {
 		}
 	}
 
+	onPaste( event ) {
+		const { title, onInsertBlockAfter, onUpdate } = this.props;
+		const clipboardData = event.clipboardData;
+
+		let plainText = '';
+		let html = '';
+
+		// IE11 only supports `Text` as an argument for `getData` and will
+		// otherwise throw an invalid argument error, so we try the standard
+		// arguments first, then fallback to `Text` if they fail.
+		try {
+			plainText = clipboardData.getData( 'text/plain' );
+			html = clipboardData.getData( 'text/html' );
+		} catch ( error1 ) {
+			try {
+				html = clipboardData.getData( 'Text' );
+			} catch ( error2 ) {
+				// Some browsers like UC Browser paste plain text by default and
+				// don't support clipboardData at all, so allow default
+				// behaviour.
+				return;
+			}
+		}
+
+		// Allows us to ask for this information when we get a report.
+		window.console.log( 'Received HTML:\n\n', html );
+		window.console.log( 'Received plain text:\n\n', plainText );
+
+		const content = pasteHandler( {
+			HTML: html,
+			plainText,
+		} );
+
+		if ( typeof content !== 'string' && content.length ) {
+			event.preventDefault();
+
+			const [ firstBlock ] = content;
+
+			if (
+				! title &&
+				( firstBlock.name === 'core/heading' ||
+					firstBlock.name === 'core/paragraph' )
+			) {
+				onUpdate( firstBlock.attributes.content );
+				onInsertBlockAfter( content.slice( 1 ) );
+			} else {
+				onInsertBlockAfter( content );
+			}
+		}
+	}
```

Comment:
Do you think there's logic to be shared between this and the other place we're using it today (RichText?)?

Reply:
I think as soon as we turn the title into a block a few things would have to be consolidated.

Reply:
Right, there is logic that can be shared, but we should instead aim to turn the post title into a block, which means the logic will automatically be shared through either RichText or PlainText v2.